Sources/Testcontainers/GenericContainer.swift

```diff
@@ -5,34 +5,35 @@ import NIO
 
 public final class GenericContainer {
     
-    let logger = Logger(label: "GenericContainer")
-    
     static let uuid = UUID().uuidString
     
-    private let name: String
+    private let logger: Logger
+    private let imageParams: ImageParams
     private let configuration: ContainerConfig
     
     private var docker: Docker?
     private var container: Docker.Container?
     private var image: Docker.Image?
     
-    public init(name: String, configuration: ContainerConfig) {
-        self.name = name
+    public init(image: ImageParams, configuration: ContainerConfig, logger: Logger? = nil) {
+        self.imageParams = image
         self.configuration = configuration
+        self.logger = logger ?? Logger(label: "GenericContainer")
         
         guard let client = DockerClientStrategy().resolve() else {
             self.docker = nil
-            logger.error("Unable to resolve a Docker client")
+            self.logger.error("Unable to resolve a Docker client")
             return
         }
         self.docker = Docker(client: client)
     }
     
-    public convenience init(name: String, port: Int) {
+    public convenience init(name: String, tag: String = "latest", port: Int) {
```

Comment:
It would be nice to be able to provide the `logger` to this init method as well, since this is one people might be using more than the first init method.

Reply:
I also want to add- since this is a convenience init, is there any way to just take the `name` param and allow you to just pass `rabbitmq:3-alpine`, and add a new init to `ImageParams` to then take this string and parse out the name and tag from it? Maybe something nice to do here to make the API clean:

`let container = GenericContainer(image: "rabbitmq:3-alpine", port: 5672, logger: logger)`